### DISCUSSION TITLE
Idea for a custom submit button on item views

### DISCUSSION THREAD

**Discussion by jschuur** [keystonejs/keystone#7836] [Feature requests]
It would be useful to have a button to trigger the 'Save Changes' process in another part of the item view layout, with the ability to recognise when that buton was used in all the usual hooks.

For my current project, one list has some properties that are pulled in from an external source. I only want these properties to be updated manually (so the logic lies outside of the usual hooks) and infrequently.

The way I do this now is to define a new segmented control select field and then in a `resolveInput` hook, if that is set to 'sync on save', I perform my custom action and reset it to 'do nothing' (the default):

<img width="517" alt="CleanShot 2022-08-23 at 09 47 10@2x" src="https://user-images.githubusercontent.com/92457/186114805-2cc25d4c-cb04-4489-93bb-56bad1dbc719.png">

This is a little clunky, because it's a two step process and the second step requires scrolling down to the end of the page, and you have to add more explicit instructions for how this works.

What if there was a new field type called `submit` that also triggers a save and sets a `resolvedInput` boolean to `true` if it was clicked?

It could just trigger the save process as normal (even if no other fields have been changed), accept all the usual field properties like `ui` and `label` (used for the button text) etc. and obey the usual hooks.

Here's how that could look like in a list definition:

```js
Config: list({
  fields: {
    // ... other list options
    sync: submit({
      label: 'Sync Data',
      ui: {
        description: 'Manually sync data from external source',
        createView: { fieldMode: 'hidden' },
        itemView: { fieldMode: ({ session }) => session.data.isEditor ? 'edit' : 'hidden',
      },
      data: json(),
    }),
    // ... other list options
    hooks: {
      resolveInput: async ({ resolvedData }) => {
        if (resolvedData.sync)
          resolvedData.data = await fetchDataFromExternalSource();

        return resolvedData;
      }
    }
  })

```

**Comment by willemmulder** [comment]
Nice idea! It would also be nice to have some UI options for placement (e.g. at the top or bottom, in flow with the other fields, etc).

I'm not sure if I always want it to perform a save though. Imagine I'm in the middle of editing a field and -just to be sure of its value- I want to update-fetch one specific field from an external source using a button. I would not want that to cause a save to the record.

But maybe that could be arranged by having a dedicated fieldtype (e.g. 'button') that would not cause a save but simply run a specific 'clicked' action. That function could also return an Object with updated fields that could then be merged on the frontend...?
